main.tf

```diff
@@ -47,6 +56,13 @@ resource "aws_iam_role_policy_attachment" "default" {
   policy_arn = join("", aws_iam_policy.default.*.arn)
 }
 
+resource "aws_iam_role_policy_attachment" "managed" {
+  for_each   = module.this.enabled ? var.managed_policy_arns : []
```

Comment:
Will this work without a set() around the list ?

Reply:
for each works only with maps or sets

Reply:
i dind't test, we can change to set()

Reply:
@nitrocode I have tested this and it works without calling toset()